src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -520,6 +520,12 @@ private static FunctionMetadata ParseFunctionMetadata(string functionName, JObje
                 functionMetadata.IsExcluded = (bool)value;
             }
 
+            if (configMetadata.TryGetValue("autoPackageRestoreEnabled", StringComparison.OrdinalIgnoreCase, out value) &&
+                value.Type == JTokenType.Boolean)
+            {
+                functionMetadata.IsAutoPackageRestoreEnabled = (bool)value;
```

Comment:
Had a discussion with @paulbatum and @lindydonna about this.
The main reasons to have this as an opt-in feature are the fact that this is primarily to support CI/CD scenarios and most users wouldn't want to have this on once the function is stable, and to serve as a safety mechanism that enables users to disable this if needed (also reduces the set of features enabled out of the box by removing something that most functions are unlikely to need).

I did go back and forth on host vs function level setting, but ended up going with a function level one since that would give users more flexibility (they could enable that for just some of the functions in the Function App)

Do you think this should be at the host level? Or that we shouldn't have the flag?